### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by senile-pitcher**

### CODE REVIEW

File: src/arrayMethodReduce.js [hunk 1]
```diff
@@ -5,7 +5,25 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    const thisArray = [...this];
+
+    if (thisArray.length === 0 && initialValue === undefined) {
+      throw new TypeError('reduce of empty array with no initial value');
+    }
```
Comment (AndriiNychaiuk) [inline src/arrayMethodReduce.js:14]: Great solution! It's easy and simple to understand!) But there is one moment. There won't be an error in the reduce method if the array is empty and the initial value is undefined. So you don't need this part in your solution I guess.

**Review by AndriiNychaiuk — CHANGES_REQUESTED** [review]

**Review by AndriiNychaiuk — APPROVED** [review]
👍

File: src/arrayMethodReduce.js [hunk 2]
```diff
@@ -5,7 +5,29 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    const thisArray = [...this];
```
Comment (G-YaSha) [inline src/arrayMethodReduce.js]: Why do need this copy? You don't mutate `this`, so you don't need copy here

File: src/arrayMethodReduce.js (hunk 1 above)
Comment (G-YaSha) [reply src/arrayMethodReduce.js:14]: Actually, It will be the error in such case, so good job👍

File: src/arrayMethodReduce.js [hunk 3]
```diff
@@ -5,7 +5,29 @@
...
+
+    if (callback === undefined) {
+      throw new TypeError('undefined is not a function');
+    }
```
Comment (G-YaSha) [inline src/arrayMethodReduce.js:10]: 👍

**Review by G-YaSha — CHANGES_REQUESTED** [review]
Almost done🔥 Great solution, let's fix some small thing

**Review by G-YaSha — APPROVED** [review]

**Pull request closed without merging**